server/middlewares/authentication.ts

```diff
@@ -146,8 +146,12 @@ export default function auth(options: AuthenticationOptions = {}) {
           getRootSpanFromRequestContext(ctx)
         );
       }
-    } else {
```

Comment:
I believe we need to keep this otherwise auth will only be set to `{}` in the event of an error

Reply:
I think the current behavior is quite the same as previously:
- If `user` is defined and has the correct permissions, `ctx.auth` will be fully set at L133-L137
- If `options.optional` is true, one exception will be thrown at some point, then caught so `ctx.auth` can be assigned to `{}` at L151
- In other cases (insufficient perms/not optional), an exception will be thrown, caught but thrown again. `ctx.auth` has indeed no value here but the API call is interrupted anyway…

Reply:
I was thinking the case where `token` is set but invalid, I suppose there the JWT parsing will fail

Reply:
Yes, that's the case!